synapse/handlers/device.py

```diff
@@ -265,6 +265,14 @@ def __init__(self, hs):
 
         hs.get_distributor().observe("user_left_room", self.user_left_room)
 
+    def _check_device_name_length(self, name):
+        if name and len(name) > MAX_DEVICE_DISPLAY_NAME_LEN:
+            raise SynapseError(
+                400,
+                "Device display name is too long (max %i)"
+                % (MAX_DEVICE_DISPLAY_NAME_LEN,),
+            )
```

Comment:
```suggestion
                errcode=Codes.TOO_LARGE,
            )
```

I'd say, rather than `M_UNKNOWN`.

Reply:
Done and updated another test to expect this also.